### PR TITLE
(FACT-1606) split fact_precedence test into 4 other tests

### PR DESCRIPTION
**Pull request by ericldelaney**
Split the 5 tests of fact_precedence.rb into 4 tests and dropped the 5th test as its covered by the options tests for external and custom facts
Applied code style, added risk tags, added TestRail numbers

### CODE REVIEW

**Comment by ericldelaney** [comment]
Tested on:
arista4-32a, ciscoxr-64a%7Bhypervisor=vmpooler%7D, cisconx-64a%7Bhypervisor=vmpooler%7D, cumulus25-64a, debian8-64a, fedora25-64a, osx1012-64a, redhat7-64a, sles12-64a, solaris11-64a, ubuntu1604-64a, windows10ent-64a, windows2016-64a

File: acceptance/tests/custom_facts/custom_fact_with_10001_weight_overrides_external_fact.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# This test verifies that a custom fact with a weight of 10001 overrides an external fact
```
Comment (kurtwall) [inline acceptance/tests/custom_facts/custom_fact_with_10001_weight_overrides_external_fact.rb]: This comment is repetitive; the `test_name` expression on the next line says the same thing.

File: acceptance/tests/external_facts/external_fact_overrides_custom_fact.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# This test verifies that facter resolves external facts over custom facts (without a weight)
```
Comment (kurtwall) [inline acceptance/tests/external_facts/external_fact_overrides_custom_fact.rb]: Maybe delete this and add _without a weight_ to the end of `test_name`?
```
test_name "C100150: external fact overrides custom fact without a weight"
```

File: acceptance/tests/external_facts/external_fact_overrides_custom_fact_with_10000_weight_or_less.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# verify that an external fact takes overrides over a custom fact of weight 10000 or less
```
Comment (kurtwall) [inline acceptance/tests/external_facts/external_fact_overrides_custom_fact_with_10000_weight_or_less.rb]: Suggest deleting the repetitive comment.

File: acceptance/tests/external_facts/external_fact_overrides_custom_fact_with_confine.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# this test verifies that an external fact overrides a custom fact with a confine
```
Comment (kurtwall) [inline acceptance/tests/external_facts/external_fact_overrides_custom_fact_with_confine.rb]: Suggest deleting the repetitive comment.

**Comment by ericldelaney** [comment]
@kurtwall Updated

**Pull request merged**